src/main/webapp/js/chart-loader.js

```diff
@@ -1,4 +1,37 @@
 (function() {
+
+  var _messageChartPromise = null;
+
+  var fetchMessageData = function() {
+    const url = "/api/chart";
+    if (_messageChartPromise == null) {
+      _messageChartPromise = fetch(url)
+        .then((response) => {
+          return response.json();
+        })
+        .then((msgJson) => {
+          var msgData = new google.visualization.DataTable();
+          //define columns for the DataTable instance
+          msgData.addColumn('date', 'Date');
+          msgData.addColumn('number', 'Message Count');
+          for (i = 0; i < msgJson.length; i++) {
+            msgRow = [];
+            var timestampAsDate = new Date(msgJson[i].timestamp);
+            var totalMessages = i + 1;
+            msgRow.push(timestampAsDate, totalMessages);
+            //TODO add the formatted values to msgRow array by using JS' push method
+            msgData.addRow(msgRow);
+          }
+          var chart = new google.visualization.BarChart(
```

Comment:
Can you move the call to DOM operations  (like `getElementById`) out of `fetchMessageData`?

Instead, there should be code like:

    `var drawChart = function() {
          fetchMessageData()
              .then(function (msgData) {
                  // ... do DOM calls like `getElementById` and `draw`
              });
       };

And then you can invoke `drawChart` in the `onload` code, while calling `fetchMessageData` earlier.